### PR TITLE
Implement --version flag

### LINKED ISSUE #207: Implement `--version` flag

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #207.

### CODE REVIEW

File: test/spec/GarnSpec.hs [hunk 1]
```diff
@@ -60,67 +61,69 @@ spec = do
...
-                type DenoOutput = {
-                  "some incompatible type": null,
-                  garnTsLibVersion: string
-                };
+      it "outputs a version with --version" $ onTestFailureLogger $ \onTestFailureLog -> do
+        output <- runGarn ["--version"] "" repoDir Nothing
+        onTestFailureLog output
+        stdout output `shouldBe` "v0.0.14\n"
+        stderr output `shouldBe` ""
+        exitCode output `shouldBe` ExitSuccess
```
Comment (soenkehahn) [inline test/spec/GarnSpec.hs:76]: The diff for this file looks terrible. It's probably easier to review with hidden white space changes. Anyways, this is the only real now test.

File: ts/internal/version.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// This is in its own file because the `with { type: "json" }` syntax breaks code formatting.
+import version from "./version.json" with { type: "json" };
```
Comment (alexdavid) [inline ts/internal/version.ts:2]: Should we consider just reading this with `Deno.readFileSync` instead if it causes issues with the formatter?

**Review by alexdavid — APPROVED** [review]
Looks good

**Comment by soenkehahn** [comment]
I realized (prompted by @alexdavid's comment) that this PR probably breaks `garn init` currently, since we don't ship that `version.json` file in the cabal install.

File: package.yaml [hunk 3]
```diff
@@ -8,6 +8,7 @@ copyright: "AllRightsReserved"
 verbatim:
   data-files: |
     ts/**/*.ts
+    ts/**/*.json
```
Comment (soenkehahn) [inline package.yaml:11]: This is using cabal globs verbatim and that doesn't allow file globs without extensions, see https://cabal.readthedocs.io/en/stable/cabal-package.html#pkg-field-data-files.

File: justfile [hunk 4]
```diff
@@ -151,7 +151,7 @@ codegen: hpack && typescript-check
   cabal run codegen
 
 typescript-check *args="":
-  deno check ts/*.ts {{ args }}
+  deno check ts/*.ts ts/**/*.ts {{ args }}
```
Comment (soenkehahn) [inline justfile:154]: Unrelated to this PR, but before we didn't typecheck files in subdirs.

File: ts/internal/version.ts (hunk 2 above)
Comment (soenkehahn) [reply ts/internal/version.ts:2]: That won't work in the case when the ts lib is imported from `https://garn.io/ts/...`. Also, it'd be hard to figure out the path when reading this from the cabal data files. I think it's easier to rely on deno's importing mechanism here.

We *do* have to make sure that the `version.json` file gets

- deployed with `just release-ts` (I checked that manually),
- included in the cabal data files (see the new change in `package.yaml`), and
- served for local development with `just fileserver` (also tested manually).

Comment (alexdavid) [reply ts/internal/version.ts:2]: Ah of course, forgot about the URL imports :confounded:

**Review by alexdavid — APPROVED** [review]

**Pull request merged**